core/src/main/java/feign/SynchronousMethodHandler.java

```diff
@@ -181,6 +183,14 @@ Object decode(Response response) throws Throwable {
     }
   }
 
+  Options findOptions(Object[] argv) {
+    if (argv == null || argv.length == 0) {
+      return this.options;
+    }
+    return Stream.of(argv).filter(o -> o instanceof Options).findAny().map(o -> (Options) o)
```

Comment:
I don't believe you need the `findAny()`.  This can be simplified:

```java
return (Options) Stream.of(argv)
         .filter(arg -> arg instanceof Options)
         .findFirst()
         .orElse(this.options);
```